### PR TITLE
IBC docs, clean up

### PR DESCRIPTION
**Pull request by tmat**
Documents IBC optimization targets and VS IBC training support.

Introduces `EnablePartialNgenOptimization` variable to control when IBC data embedding is performed.

### CODE REVIEW

**Comment by tmat** [comment]
@jmarolf @livarcocc PTAL

File: Documentation/ArcadeSdk.md [hunk 1]
```diff
@@ -539,6 +556,129 @@ This step is required for repositories that build VS insertion components.
   condition: succeeded()
 ```
 
+## IBC Optimization Data Embedding
+
+The Arcade SDK includes targets that enable IBC optimization data to be embedded to assemblies produced by the repository.
+
+To enable this functionality set `UsingToolIbcOptimization` to `true` in `/eng/Versions.props`.
+
+Typically, not all projects in a repository need IBC data embedded. Set `ApplyPartialNgenOptimization` is `true` in a project to indicate that the assemblies produced by the project should get IBC data embedded.
```
Comment (jcagme) [inline Documentation/ArcadeSdk.md]: >is [](start = 103, length = 2)

is -> to

File: Documentation/ArcadeSdk.md [hunk 2]
```diff
@@ -539,6 +556,129 @@ This step is required for repositories that build VS insertion components.
...
+
+Use `EnablePartialNgenOptimization` property to control when IBC data embedding is gonna be performed. Unless specified otherwise, `EnablePartialNgenOptimization` is set to `true` if `Configuration` is `Release` and `OfficialBuild` is `true`.
+
+The IBC data embedding is performed by an internal tool `ibcmerge.exe` provided by `Microsoft.Dotnet.IbcMerge` package from an internal Azure DevOps feed. The repository build definition thus must invoke Azure DevOps task that restores internal tools in order for IBC data embedding to work. See [Restoring internal tools](#restoring-internal-tools).
+
+Unless the repository is using an IBC data acquision mechanism built into the Arcade SDK (such as [Visual Studio IBC Data Acquisition](#visual-studio-ibc-data-acquisition)) the repository must set the value of `IbcOptimizationDataDir` property and its build must ensure that IBC data are present at location specified this property before the `Build` target is executed. It is recommended that such logic is implemented in `/eng/Tools.props` via a target chained after the `Restore` target.
```
Comment (jcagme) [inline Documentation/ArcadeSdk.md]: >at [](start = 296, length = 2)

at -> at the

Comment (jcagme) [inline Documentation/ArcadeSdk.md]: >this [](start = 318, length = 4)

this  -> by this (?)

File: Documentation/ArcadeSdk.md [hunk 3]
```diff
@@ -539,6 +556,129 @@ This step is required for repositories that build VS insertion components.
...
+
+The IBC data drop produced by a training run is identitifed by the name of the repository, the branch and the build number the trained binaries came from, and a training run id. An example of IBC data identifier is `OptimizationData/dotnet/roslyn/master-vs-deps/20190210.1/935479/1`, where `dotnet/roslyn` is the repository name, `master-vs-deps` is the branch name, `20190210.1` is the build number and `935479/1` is training run id. The IBC data acquisition implementation in Arcade SDK requires `RepositoryName` and `VisualStudioIbcSourceBranchName` properties to be set globally or in `/eng/Tools.props` file. These properties specify the repository name and the branch name to use to find the IBC data to acquire. Set `VisualStudioIbcDropId` to the combination of build number and training run id (e.g. `20190210.1/935479/1`) to indicate the exact IBC data drop to acquire. If `VisualStudioIbcDropId` is not specified the most recent IBC data published from the specified repository and branch are acquired.
+
+The IBC data are downloaded to `IbcOptimizationDataDir`, which is initialized to `.tools/IbcData`.
+
+### Visual Studio Training Inputs Generation
+
+The training process requires
```
Comment (jcagme) [inline Documentation/ArcadeSdk.md]: >s [](start = 28, length = 1)

Add ':'

File: Documentation/ArcadeSdk.md [hunk 4]
```diff
@@ -482,6 +482,23 @@ The steps below assume the following variables to be defined:
   condition: and(succeeded(), ne(variables['SignType'], ''))
 ```
 
+### Restoring internal tools
```
Comment (jcagme) [inline Documentation/ArcadeSdk.md]: It'd be good if lines could be shorter so there is no too much scrolling right/left when reviewing in CodeFlow. I think breaking between column 100-110 should do the trick

**Comment by tmat** [comment]
@jcagme Thanks!

**Pull request merged**